x/rollapp/keeper/msg_server_create_rollapp.go

```diff
@@ -45,21 +45,20 @@ func (k msgServer) CreateRollapp(goCtx context.Context, msg *types.MsgCreateRoll
 }
 
 func (k msgServer) checkIfRollappExists(ctx sdk.Context, msg *types.MsgCreateRollapp) error {
-	rollappId, err := types.NewChainID(msg.RollappId)
-	if err != nil {
-		return err
-	}
+	rollappId, _ := types.NewChainID(msg.RollappId)
 	// check to see if the RollappId has been registered before
 	if _, isFound := k.GetRollapp(ctx, rollappId.GetChainID()); isFound {
 		return types.ErrRollappExists
 	}
-	if !rollappId.IsEIP155() {
-		return nil
-	}
-	// check to see if the RollappId has been registered before with same key
+
+	// check to see if the RollappId has been registered before with same EIP155 ID
 	existingRollapp, isFound := k.GetRollappByEIP155(ctx, rollappId.GetEIP155ID())
 	// allow replacing EIP155 only when forking (previous rollapp is frozen)
 	if !isFound {
+		// if not forking, check to see if the Rollapp has been registered before with same name
+		if _, isFound = k.FindRollappByName(ctx, rollappId.GetName()); isFound {
+			return types.ErrRollappExists
```

Comment:
might wanna add a reason to this error

Reply:
with errorsmod.Wrap

Reply:
also the description of the ErrRollappExists specificiall refers to ID, not name, so it's misleading

Reply:
imo `return errorsmod.Wrap(gerrc.ErrAlreadyExists, "name")`